### PR TITLE
E2E Tests: fix user creation and switching

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Context

As we've discovered, the non-admin e2e tests were not actually running as expected.

Besides the users not actually having been created initially, there was also an issue with not the right user being logged in and not detecting that in the tests.

## Summary

Fixes user creation in the local environment setup.

Also makes e2e tests for non-admins more robust. See #7807.

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [ ] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [ ] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [ ] I have added a matching `Type: XYZ` label to the PR

---

See #7790

### CODE REVIEW

File: bin/local-env/install-wordpress.sh [hunk 1]
```diff
@@ -76,6 +76,7 @@ if [ ! "$(wp user get author --field=login)" ]; then
 else
  echo -e $(status_message "Author already exists, skipping...")
 fi
+wp user get contributor --field=login
```
Comment (swissspidy) [inline bin/local-env/install-wordpress.sh]: The user hasn't been created yet at this point, so this errors and breaks the setup. Need to move it further down.

Comment (spacedmonkey) [reply bin/local-env/install-wordpress.sh]: this is debug

Comment (spacedmonkey) [reply bin/local-env/install-wordpress.sh]: See this line for context - https://github.com/google/web-stories-wp/runs/2728765491?check_suite_focus=true#step:13:664

Comment (swissspidy) [reply bin/local-env/install-wordpress.sh]: Here's the line where it breaks: https://github.com/google/web-stories-wp/pull/7803/checks?check_run_id=2728998575#step:13:645

Comment (swissspidy) [reply bin/local-env/install-wordpress.sh]: Now I see why you added that debug statement though!

2bf16c69d5587316794de73404cfaeb45354062c0cd975a6223f47ce682221ba

Comment (spacedmonkey) [reply bin/local-env/install-wordpress.sh]: The contributor and other users are not being creatd for some reason, so debug

Comment (swissspidy) [reply bin/local-env/install-wordpress.sh]: So either the user is not actually created (which is a bug in the setup script), _or_ the `loginUser` function doesn't properly type in the username.

Running `wp user list` for debugging here would be helpful as a first step

File: packages/e2e-tests/src/specs/wordpress/getStartedStory.js [hunk 2]
```diff
@@ -62,6 +62,7 @@ describe('Get Started Story', () => {
         'post-new.php',
         'post_type=web-story&web-stories-demo=1'
       );
+      await percySnapshot(page, 'post title pre-fill');
```
Comment (swissspidy) [inline packages/e2e-tests/src/specs/wordpress/getStartedStory.js]:
```suggestion
      await percySnapshot(page, 'Get Started Story (Author)');
```

**Comment by swissspidy** [comment]
@googlebot I consent

File: packages/e2e-tests/src/specs/editor/authorUser.js [hunk 3]
```diff
@@ -55,6 +56,7 @@ describe('Author User', () => {
 
   it('should be able to directly preview a story without markup being stripped', async () => {
     await createNewStory();
+    await percySnapshot(page, 'Stripped markup');
```
Comment (spacedmonkey) [inline packages/e2e-tests/src/specs/editor/authorUser.js]: Remove this is debug

File: packages/e2e-tests/src/specs/wordpress/getStartedStory.js (hunk 2 above)
Comment (spacedmonkey) [reply packages/e2e-tests/src/specs/wordpress/getStartedStory.js]: Remove this is debug

File: packages/e2e-tests/src/specs/wordpress/quickEdit.js [hunk 4]
```diff
@@ -38,6 +39,7 @@ describe('Quick Edit', () => {
 
   it('should save story without breaking markup', async () => {
     await createNewStory();
+    await percySnapshot(page, 'breaking markup');
```
Comment (spacedmonkey) [inline packages/e2e-tests/src/specs/wordpress/quickEdit.js]: Remove this is debug

File: bin/local-env/install-wordpress.sh [hunk 5]
```diff
@@ -64,31 +64,35 @@ fi
...
 fi
-if [ ! "$(wp user get subscriber --field=login)" ]; then
+if [[ $(wp user get subscriber --field=login 2>&1) != "subscriber" ]]; then
 	wp user create subscriber subscriber@example.com --role=subscriber --user_pass=password --quiet
 	echo -e $(status_message "Subscriber created! Username: subscriber Password: password")
 else
  echo -e $(status_message "Subscriber already exists, skipping...")
 fi
 
+wp user list
```
Comment (spacedmonkey) [inline bin/local-env/install-wordpress.sh]: @swissspidy I think we should leave this here. It is a good way to see in logs if user creation. Maybe we could format it look nicer in the logs..

I am also thinking about adding wp media list after attachments are made, so we can easily debug.

**Review by swissspidy — APPROVED** [review]

**Comment by spacedmonkey** [comment]
@swissspidy Make some tweaks. Give me a minute!

File: packages/e2e-test-utils/src/activatePlugin.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+import { setCurrentUser } from './user';
+import visitAdminPage from './visitAdminPage';
+
+async function activatePlugin(slug) {
+  await setCurrentUser('admin', 'password');
+  await visitAdminPage('plugins.php');
+  const disableLink = await page.$(`tr[data-slug="${slug}"] .deactivate a`);
+
+  if (disableLink) {
+    await setCurrentUser('admin', 'password');
```
Comment (swissspidy) [inline packages/e2e-test-utils/src/activatePlugin.js]: That does nothing..?

The user is already an admin at this point.

I think what we'd want in this util is:

```js
const currentUser = getCurrentUser();
await setCurrentUser('admin', 'password');

if ( can activate plugin ) {
  // activate plugin
}

await setCurrentUser(currentUser.username, currentUser.password);
```

File: packages/e2e-test-utils/src/createNewPost.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+ * Creates new post.
+ *
+ * @param {Object}  object                    Object to create new post, along with tips enabling option.
+ * @param {string}  [object.postType]         Post type of the new post.
+ * @param {string}  [object.title]            Title of the new post.
+ * @param {string}  [object.content]          Content of the new post.
+ * @param {string}  [object.excerpt]          Excerpt of the new post.
+ * @param {boolean} [object.showWelcomeGuide] Whether to show the welcome guide.
+ */
+async function createNewPost({
```
Comment (swissspidy) [inline packages/e2e-test-utils/src/createNewPost.js:35]: The only difference is the usage of our `visitAdminPage` util?

Comment (spacedmonkey) [reply packages/e2e-test-utils/src/createNewPost.js:35]: Yes

File: packages/e2e-test-utils/src/deactivatePlugin.js [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+import { setCurrentUser } from './user';
+import visitAdminPage from './visitAdminPage';
+
+async function deactivatePlugin(slug) {
+  await setCurrentUser('admin', 'password');
+  await visitAdminPage('plugins.php');
+  const deleteLink = await page.$(`tr[data-slug="${slug}"] .delete a`);
+
+  if (deleteLink) {
+    await setCurrentUser('admin', 'password');
```
Comment (swissspidy) [inline packages/e2e-test-utils/src/deactivatePlugin.js]: Ditto here about `setCurrentUser` usage.

File: packages/e2e-test-utils/src/activatePlugin.js (hunk 6 above)
Comment (spacedmonkey) [reply packages/e2e-test-utils/src/activatePlugin.js]: Copied from 
https://github.com/WordPress/gutenberg/blob/2826f5005df7f8e303730dc79aded9e5ea890fd5/packages/e2e-test-utils/src/deactivate-plugin.js#L16-L20

I dont really know what this does.

Comment (swissspidy) [reply packages/e2e-test-utils/src/activatePlugin.js]: In Gutenberg that does:

1. Switch to admin
2. Go to plugins page
3. -> If plugin is already deactivated, switch back to user being tested, and do nothing
4. Deactivate plugin
4. Switch back to user being tested

Similar for plugin activation.

So what we want is roughly what I pasted above, essentially the same logic.

Just not two `setCurrentUser('admin', 'password')` in a row.

Comment (swissspidy) [reply packages/e2e-test-utils/src/activatePlugin.js]:
```js
/**
 * Internal dependencies
 */
import { getCurrentUser, setCurrentUser } from './user';
import visitAdminPage from './visitAdminPage';

async function activatePlugin(slug) {
  const currentUser = getCurrentUser();
  await setCurrentUser('admin', 'password');

  await visitAdminPage('plugins.php');
  const disableLink = await page.$(`tr[data-slug="${slug}"] .deactivate a`);

  if (!disableLink) {
    await page.click(`tr[data-slug="${slug}"] .activate a`);
    await page.waitForSelector(`tr[data-slug="${slug}"] .deactivate a`);
  }

  await setCurrentUser(currentUser.username, currentUser.password);
}

export default activatePlugin;
```

**Pull request merged**